experimental/PropertyDDS/packages/property-changeset/src/utils.ts

```diff
@@ -2,2261 +2,2097 @@
+         */
+        postCallback?: (context: TraversalContext) => any,
+        /**
+         * An optional object that is passed to all invocations of the callback via the
+         */
+        userData?: { [key: string]: any },
+        /**
+         *  The operation that has been applied to the root of the ChangeSet (either 'insert' or 'modify')
+         */
+        rootOperation?: OperationType,
+        /**
+         *  The full typeid for the Property at the root of the ChangeSet
+         */
+        rootTypeid?: string,
+    }
 
-/**
- * Stop the traversal for all nodes below the currently processed one
- * @private
- */
-Utils.TraversalContext.prototype.stopTraversal = function() {
-    this._traversalStopped = true;
-};
+    /**
+        * Traverses a ChangeSet recursively and invokes the callback for each visited property.
+        *
+        * @param in_preCallback  - The (pre-order) callback function that is invoked for each property
+        * @param in_postCallback - The (post-order) callback function that is invoked for each property
+        * @param in_context -  The traversal context for the currently processed property
+        * @param in_levelCallback - A callback for when a node is reached
+        */
+    function _traverseChangeSetRecursivelyAsync(
+        in_preCallback: (
+            context: TraversalContext,
+            next: (err?: Error | null | undefined | string, result?: unknown) => void) => any,
+        in_postCallback: (
+            context: TraversalContext,
+            next: (err?: Error | null | undefined | string, result?: unknown) => void) => any,
+        in_context: TraversalContext,
+        in_levelCallback: (param?: any) => any) {
+        let pathSeparator;
+        let currentPath;
+        let currentPostPath;
+        let nestedChangeSet;
+        let postOrderContext;
+
+        // Call the callback function for this ChangeSet
+        in_context.startTraversal();
+        const typeid = in_context.getTypeid()
+        let splitTypeId = typeid !== undefined ?
+            TypeIdHelper.extractContext(typeid) :
+            undefined;
+        in_context.setSplitTypeID(splitTypeId!);
+
+        let currentUserData;
+
+        series([
+
+            function(next: NextFn) {
+                if (in_preCallback !== undefined) {
+                    in_preCallback(in_context, next);
+                } else {
+                    next();
+                }
+            },
+
+            function(next: NextFn) {
+                pathSeparator = in_context.getFullPath() !== "" ? PROPERTY_PATH_DELIMITER : "";
+                currentPath = in_context.getFullPath();
+                currentPostPath = in_context.getFullPostPath();
+                nestedChangeSet = in_context.getNestedChangeSet();
+
+                const _typeid = in_context.getTypeid();
+                // Call the callback function for this ChangeSet
+                in_context.startTraversal();
```

Comment:
in a lot of places in this function, you replaced direct member access by calling a setter / getter. These are quite performance critical functions. Do you think this will have a performance impact?

Reply:
Probably yet, but I would assume its very minor. Do you think we should revert that back ?